go/plugin/agentcard-rewriter/agentcard_rewriter.go

```diff
@@ -0,0 +1,147 @@
+var logger = logging.New(pluginName)
+
+func main() {}
+
+func init() {
+	logger.Info("loaded")
+}
+
+// RegisterHandlers registers the plugin with KrakenD
+func (r registerer) RegisterHandlers(f func(
+	name string,
+	handler func(context.Context, map[string]interface{}, http.Handler) (http.Handler, error),
+)) {
+	f(string(r), r.registerHandlers)
+	logger.Info("registered")
+}
+
+// responseWriter wraps http.ResponseWriter to capture response body
+type responseWriter struct {
+	http.ResponseWriter
+	body       *bytes.Buffer
+	statusCode int
+}
+
+func newResponseWriter(w http.ResponseWriter) *responseWriter {
+	return &responseWriter{
+		ResponseWriter: w,
+		body:           &bytes.Buffer{},
+		statusCode:     http.StatusOK,
+	}
+}
+
+func (rw *responseWriter) Write(b []byte) (int, error) {
+	return rw.body.Write(b)
+}
+
+func (rw *responseWriter) WriteHeader(statusCode int) {
+	rw.statusCode = statusCode
+}
+
+func (r registerer) registerHandlers(_ context.Context, extra map[string]interface{}, handler http.Handler) (http.Handler, error) {
+	logger.Info("plugin initialized successfully")
+	return http.HandlerFunc(r.handleRequest(handler)), nil
+}
+
+func (r registerer) handleRequest(handler http.Handler) func(w http.ResponseWriter, req *http.Request) {
+	return func(w http.ResponseWriter, req *http.Request) {
+		reqLogger := logging.NewWithPluginName(pluginName)
+
+		// Check if this is a GET request to an agent card endpoint
+		if req.Method == http.MethodGet && isAgentCardEndpoint(req.URL.Path) {
+			reqLogger.Debug("intercepted agent card request: %s", req.URL.Path)
+
+			// Get gateway URL
+			gatewayURL, err := getGatewayURL(req)
+			if err != nil {
+				reqLogger.Error("cannot determine gateway URL: %s", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			// Extract full agent path from request (everything before /.well-known)
+			agentPath := extractAgentPath(req.URL.Path)
+			if agentPath == "" {
+				reqLogger.Warn("cannot extract agent path from: %s - passing through", req.URL.Path)
+				handler.ServeHTTP(w, req)
+				return
+			}
+
+			reqLogger.Debug("rewriting URLs for agent path: %s, gateway: %s", agentPath, gatewayURL)
+
+			// Wrap response writer to capture backend response
+			rw := newResponseWriter(w)
+
+			// Forward request to backend
+			handler.ServeHTTP(rw, req)
+
+			// Only transform successful responses
+			if rw.statusCode != http.StatusOK {
+				reqLogger.Info("backend returned non-OK status: %d - returning error", rw.statusCode)
+				http.Error(w, "Backend service returned an error", rw.statusCode)
+				return
+			}
+
+			// Validate content type
+			contentType := rw.Header().Get("Content-Type")
+			if !strings.Contains(contentType, "application/json") {
+				reqLogger.Warn("unexpected content-type: %s - returning error", contentType)
+				http.Error(w, "Expected application/json content type", http.StatusInternalServerError)
```

Comment:
That's not an internalServerError. There should be a suitable 4xx error, otherwise 400 is ok.